File: leetcode/POTD/2025-02-19/solution.java
class Solution {
    public String getHappyString(int n, int k) {
        List<String> res = new ArrayList<>();

        generate("", n, res);
        return res.size() < k ? "" : res.get(k - 1);
    }

    private void generate(String s, int n, List<String> res) {
        if (s.length() == n) {
            res.add(s);
            return;
        }
        for (char ch : new char[]{'a', 'b', 'c'}) {
            if (s.isEmpty() || s.charAt(s.length() - 1) != ch) {
                generate(s + ch, n, res);
            }
        }
    }
}

File: leetcode/POTD/2025-02-19/solution.rs
impl Solution {
    pub fn get_happy_string(n: i32, k: i32) -> String {
        let mut res = Vec::new();
        Solution::generate("".to_string(), n as usize, &mut res);
        res.get((k - 1) as usize).cloned().unwrap_or_else(|| "".to_string())
    }

    fn generate(s: String, n: usize, res: &mut Vec<String>) {
        if s.len() == n {
            res.push(s);
            return;
        }
        for ch in ['a', 'b', 'c'] {
            if s.is_empty() || s.chars().last().unwrap() != ch {
                Solution::generate(format!("{}{}", s, ch), n, res);
            }
        }
    }
}
